### ISSUE TITLE
(Some?) boolean options in /modules/widgets/icon-tasks.nix don't get converted to strings

### ISSUE THREAD

**Issue by ada4a** [nix-community/plasma-manager#253]
I was trying to disable scroll-to-switch-between-tasks. But:
- Setting the mid-level boolean option `config.General.wheelEnabled = false` causes the following error:
  ``` 
  A definition for option `[...].wheelEnabled` is not of type `string or list of string`
  ```
  so I changed it to `config.General.wheelEnabled = "false"`, which worked (which is weird, because this was supposed to be a boolean option, no?).
  
  This contradicts [this answer](https://github.com/nix-community/plasma-manager/issues/223#issuecomment-2184940524), so maybe something changed since then that broke this?
- Initially, I tried to use the high-level boolean option `config.General.behavior.wheel.switchBetweenTasks = false`. But I get a similar error:
  ```
  A definition for option `[...].behavior` is not of type `string or list of string`
  ```
  Now this is harder to solve (I think even impossible as it is now?).
  Normally the option `wheel.switchBetweenTasks = false` turns into the above mentioned mid-level option:
  https://github.com/nix-community/plasma-manager/blob/995d818078778b366e6302ea32d83c2ba586e015/modules/widgets/icon-tasks.nix#L125
  But to make everything build, you'd need to have written something like `config.General.behavior = ["wheel.switchBetweenTasks = false"];`. But will be recognized as a string, copied into the real config file verbatim (I think?), and cause an error.

Now that I think about it, these two errors are kind of separate? It may make sense to open a second issue.

**Comment by magnouvean** [comment]
At the moment the `config` option for widgets only support strings, so the answer linked won't work right now (that's an error on my part). With that said it will be possible very soon with #248. The reason the `icon-tasks` module works is because `behavior.wheel.switchBetweenTasks` (a variable created using `mkBoolOption`) will be automatically converted to "true" or "false" (i.e. a string) due to the way `mkBoolOption` is constructed. This whole system was kind of a mess, but it will, as said, soon see refactoring improvements :)

**Comment by ada4a** [comment]
I see, thanks! And it turns out I got a little bit confused myself. The option `iconTasks.behavior.wheel.switchBetweenTasks` does exist and work as intended, but it's a part of the new `icon-tasks` module. The reason I didn't use it was that I was getting errors about the module not existing, but that was due to me having an old version of plasma-manager. I updated it, and now everything works perfectly :)
